queries/scala/function_body.scm

```diff
@@ -0,0 +1,4 @@
+(function_declaration
+  body: (block) @body)
+(function_definition
+  body: (block) @body)
```

Comment:
```suggestion
(function_definition
  body: (_) @body)
(lambda_expression
  parameters: (_)
  (_) @body)
```

Removed `function_declaration` (they don't have body), and added lambda.
Also use wildcards `(_)`, because a function body is not always a `block` in scala (it can be an `indented_block`, an `infix_expression`, etc). You can see all the different cases playing around with [treesitter playground](https://github.com/nvim-treesitter/playground)